backend-rails/spec/models/service_order_spec.rb

```diff
@@ -31,4 +31,86 @@
   context 'inclusion' do
     it { should validate_inclusion_of(:status).in_array(ServiceOrder::STATUSES) }
   end
+
+  describe '#products' do
+    let(:service_order) { create(:service_order) }
+    let(:products) { service_order.products }
+
+    context 'when service order does not have a product' do
+      it 'returns an empty array' do
+        expect(products).to eq([])
+      end
+    end
+
+    context 'when service order has a product' do
+      let(:item) { create(:item) }
+      let(:item_service) { create(:item, name: 'Setup softwares', extra_info: 'nice softwares', item_type: 'service') }
+      let!(:item_service_order) { create(:item_service_order, item_id: item.id, service_order_id: service_order.id) }
```

Comment:
do you really need to use the entity `item_service_order`?
With a quick look I think you don't, but I am not sure

Reply:
I'll look at it carefully, I'm also not sure.